src/migratus/cli.clj

```diff
@@ -95,17 +98,109 @@
 
       :else (no-match-message args summary))))
 
+(defn util-date-to-local-datetime [util-date]
+  (if (= util-date nil)
+    nil
+    (let [instant (.toInstant util-date)
+          zone-id (ZoneId/systemDefault)
+          local-datetime (.atZone instant zone-id)]
+      local-datetime)))
+
+(defn formatted-date [date]
+  (if (nil? date)
+    nil
+    (core/format "%1tFT%<tTZ", date)))
+
+(defn parsed-migrations-data [cfg]
+  (let [all-migrations (migratus/all-migrations cfg)]
+    (map (fn [m] (let [{:keys [id name applied]} m
+                       to-local-date (if (nil? applied)
+                                       nil (util-date-to-local-datetime applied))
+                       date-str (formatted-date to-local-date)]
+                   {:id id :name name :applied date-str})) all-migrations)))
+
+(defn pending-migrations [cfg]
+  (filter (fn [mig] (= nil (:applied mig))) (parsed-migrations-data cfg)))
+
+(defn applied-migrations [cfg]
+  (filter (fn [mig] (not= nil (:applied mig))) (parsed-migrations-data cfg)))
+
+(defn cli-print-migrations-edn! [data]
+  (log/info data))
+
+(defn cli-print-migrations-json! [data]
+  (log/info (cheshire/generate-string data)))
+
+(defn cli-print-migrations! [data f]
+  (case f
+    "edn" (cli-print-migrations-edn! data)
+    "json" (cli-print-migrations-json! data)
+    nil))
+
+(defn col-width
+  "Set column width for CLI table"
+  [n]
+  (apply str (repeat n "-")))
+
+(defn table-line [n]
+  (let [str (str "%-" n "s")]
+    (core/format str, (col-width n))))
+
+
+(defn mig-print-fmt [data]
+  (log/info (table-line 67))
+  (log/info (core/format "%-18s%-25s%-22s%s",
+                         "| MIGRATION-ID" "| NAME" "| APPLIED" " |"))
+  (log/info (table-line 67))
+  (doall
```

Comment:
Use doseq instead of doall map .

doseq is used for
> Repeatedly executes body (presumably for side-effects) 
 https://clojuredocs.org/clojure.core/doseq

Reply:
changed (doall (map ...))  with doseq.